sdk/cosmos/azure-cosmos/docs/StatusCodes.md

```diff
@@ -0,0 +1,53 @@
+|410|21006|x|`Global strong write barrier not met` - Indicates that synchronous replication of a write operation in a multi-region account with strong consistency did not complete. This error should always be transient and will be mapped to a CosmosException with status code `503` when surfacing it to the application after exceeding SDK-retries.|
+|410|21007|x|`Read quorum not met` - Indicates that no read quorum could be achieved when using strong or bounded staleness consistency. This error should always be transient and will be mapped to a CosmosException with status code `503` when surfacing it to the application after exceeding SDK-retries.|
+|410|21007|x|`Read quorum not met` - Indicates that no read quorum could be achieved when using strong or bounded staleness consistency. This error should always be transient and will be mapped to a CosmosException with status code `503` when surfacing it to the application after exceeding SDK-retries.|
+|412|0||`Precondition failed` - The document has been modified since the application read it (and retrieved the etag that was used as pre-codnition for the write operation). This is the typical optimistic concurrency signal - and needs to be gracefully handled in your application. The usual patterns is to re-read the document, apply teh same changes and retry the write with the updated etag.|
+|429|3200||`User throttling` - Indicates that the operations being processed by your Cosmso DB account exceed the provisioned throughptu RU/s. Mitigation can be done by either scaling-up - or improving the efficiency especially of queries to reduce the RU/s consumption. See [TSG](https://learn.microsoft.com/azure/cosmos-db/nosql/troubleshoot-request-rate-too-large) for more details.|
+|429|3201||`Metadata throttling` - Indicates that metadata operations are being throttled. Increasing provisioned throughput (RU/s) won't help - this usually indicates a bug in your application where meatdata calls are triggered extensively or you are not using a singleton pattern for `CosmosClinet`/`CosmosAsyncClient`. See [TSG](https://learn.microsoft.com/azure/cosmos-db/nosql/troubleshoot-request-rate-too-large) for more details.|
+|429|< 3200|x||`SLA violating throttling` - Indicates service-side throttling that will count against the service's SLA. These errors should always be transient and will surface as a CosmosException with status code `503` after exceeding SDK-retries| 
+|449|0|x||`RetryWith` - Indicates a concurrent attempt to chaneg documents server-side - for example via patch or stored procedure invocation. The `449` status code will be automatically retried by the SDK. This condition should always be transient as long as the application is not excessively doing concurrent changes to documents.|
+|500|0||`Internal Server error` - Indicates unexpected and unqualified internal service error. This condition should always be transient.|
```

Comment:
500 may not be always transient, could be due to a bug?

Reply:
Changed in next iteration